src/core/stats.h

```diff
@@ -22,23 +22,57 @@
 #include <vector>
 
 #include "core/options.h"
+#include "core/util.h"
 
 namespace pyston {
 
+#define SLOWPATH_TIMERS 1
+#define STAT_ALLOCATIONS 0
+
+
 #define DISABLE_STATS 0
 
 #if !DISABLE_STATS
+#define STAT_TIMER(id, name)                                                                                           \
+    static int _stid##id = Stats::getStatId(name);                                                                     \
+    StatTimer _st##id(_stid##id)
+#define STAT_TIMER2(id, name, at_time)                                                                                 \
+    static int _stid##id = Stats::getStatId(name);                                                                     \
+    StatTimer _st##id(_stid##id, at_time)
+#else
+#define STAT_TIMER(id, name) StatTimer _st##id(0);
+#define STAT_TIMER2(id, name, at_time) StatTimer _st##id(0, at_time);
+#endif
+
+#define STAT_TIMER_NAME(id) _st##id
+
+#if !DISABLE_STATS && SLOWPATH_TIMERS
+#define SLOWPATH_TIMER(id, name) STAT_TIMER(id, name)
+#define SLOWPATH_TIMER2(id, name, at_time) STAT_TIMER2(id, name, at_time)
+#else
+#define SLOWPATH_TIMER(id, name)
+#define SLOWPATH_TIMER2(id, name, at_time)
+#endif
```

Comment:
just a thought, but should we combine SLOWPATH_TIMER and STAT_TIMER?  some of the timers we probably want all the time (as opposed to the slowpath ones which seem to add some overhead), but it seems like it's tricky to remember to call the right type of timer.